File: clients/csharp/Exceptions/AuthenticationException.cs
namespace LimyeDB.Exceptions;

/// <summary>
/// Thrown when authentication with the LimyeDB server fails.
/// </summary>
public class AuthenticationException : LimyeDBException
{
    public AuthenticationException(string message) : base(message, 401)
    {
    }

    public AuthenticationException(string message, Exception innerException) : base(message, 401, innerException)
    {
    }
}

File: clients/java/src/main/java/com/limyedb/exceptions/AuthenticationException.java
package com.limyedb.exceptions;

/**
 * Thrown when authentication with the LimyeDB server fails.
 */
public class AuthenticationException extends LimyeDBException {

    public AuthenticationException(String message) {
        super(message, 401);
    }

    public AuthenticationException(String message, Throwable cause) {
        super(message, 401, cause);
    }
}
